RecoMuon/MuonIdentification/plugins/MuonIdProducer.cc

```diff
@@ -1004,149 +953,148 @@ void MuonIdProducer::fillArbitrationInfo( reco::MuonCollection* pOutputMuons )
-		  if( segmentIter1->cscSegmentRef.isNonnull() && segmentIter2->cscSegmentRef.isNonnull() )
-		    if( meshAlgo_->isDuplicateOf(segmentIter1->cscSegmentRef,segmentIter2->cscSegmentRef) &&
-			(segmentIter2->mask & 0x1e0000) &&
-			(segmentIter1->mask & 0x1e0000) )
-		      segmentIter2->setMask(reco::MuonSegmentMatch::BelongsToTrackByME1aClean);
-		  //if the track has lost the segment already through normal arbitration no need to do it again.
-		}
-	    }// mark all ME1/a duplicates that this track owns
-
-         } // segmentIter1
-
-         // chamber segment sort
-         if(chamberPairs.empty()) continue; // this should never happen
-         if(chamberPairs.size()==1) {
-            chamberPairs.front().second->setMask(reco::MuonSegmentMatch::BestInChamberByDRSlope);
-            chamberPairs.front().second->setMask(reco::MuonSegmentMatch::BestInChamberByDXSlope);
-            chamberPairs.front().second->setMask(reco::MuonSegmentMatch::BestInChamberByDR);
-            chamberPairs.front().second->setMask(reco::MuonSegmentMatch::BestInChamberByDX);
-         } else {
-            sort(chamberPairs.begin(), chamberPairs.end(), SortMuonSegmentMatches(reco::MuonSegmentMatch::BestInChamberByDRSlope));
-            chamberPairs.front().second->setMask(reco::MuonSegmentMatch::BestInChamberByDRSlope);
-            sort(chamberPairs.begin(), chamberPairs.end(), SortMuonSegmentMatches(reco::MuonSegmentMatch::BestInChamberByDXSlope));
-            chamberPairs.front().second->setMask(reco::MuonSegmentMatch::BestInChamberByDXSlope);
-            sort(chamberPairs.begin(), chamberPairs.end(), SortMuonSegmentMatches(reco::MuonSegmentMatch::BestInChamberByDR));
-            chamberPairs.front().second->setMask(reco::MuonSegmentMatch::BestInChamberByDR);
-            sort(chamberPairs.begin(), chamberPairs.end(), SortMuonSegmentMatches(reco::MuonSegmentMatch::BestInChamberByDX));
-            chamberPairs.front().second->setMask(reco::MuonSegmentMatch::BestInChamberByDX);
+               if ( !muon2.isTrackerMuon() ) continue;
+               // chamberIter2
+               for ( auto& chamber2 : muon2.matches() )
+               {
+                 // segmentIter2
+                 for ( auto& segment2 : chamber2.segmentMatches )
+                 {
+                   if(segment2.isMask()) continue; // has already been arbitrated
+                   if(fabs(segment2.x       - segment1.x      ) < 1E-3 &&
+                      fabs(segment2.y       - segment1.y      ) < 1E-3 &&
+                      fabs(segment2.dXdZ    - segment1.dXdZ   ) < 1E-3 &&
+                      fabs(segment2.dYdZ    - segment1.dYdZ   ) < 1E-3 &&
+                      fabs(segment2.xErr    - segment1.xErr   ) < 1E-3 &&
+                      fabs(segment2.yErr    - segment1.yErr   ) < 1E-3 &&
+                      fabs(segment2.dXdZErr - segment1.dXdZErr) < 1E-3 &&
+                      fabs(segment2.dYdZErr - segment1.dYdZErr) < 1E-3)
```

Comment:
Previously, you changed fabs() to std::abs(), but here you don't make a change. Why change it in one place in the file and not another? 
Also, it would be better to write a function that performs this check for approximate equality:

bool approxEqual(const double value1, const double value2) {
return (std::abs(value1 - value2) < 1E-3);
}